### ISSUE TITLE
WorkingDirectory is not honored in validation

### ISSUE THREAD

**Issue by jfernandeze** [cake-contrib/Cake.Karma#7]
We have a project structure in which the build.cake file is at the root of the repo.
And we have the Angular app at a path inside "src/WebSpa"

We have checked that in command line, karma works when called from src/WebSpa
To do that in cake, we used WorkingDirectory property of KarmaStartSettings, passing "src/WebSpa"
But if we strip "src/WebSpa" from ConfigFile ("ClientApp/test/karma.conf.js") ValidateSettings fails, because it looks for configfile at the location of build.cake, not honoring WorkingDirectory.
If we leave there the full path, ValidateSettings pass, but then Cake makes the path absolute, passing karma something like /src/WebSpa/src/WebSpa/ClientApp/test/karma.conf.js, which clearly isn't there

So I guess, in ValidateSettings, instead of

if (!FileSystem.Exist(settings.ConfigFile))

the path that should be checked should be possibly modified by WorkingDirectory

Thanks a lot, by the way

**Comment by RaringCoder** [comment]
Sorry, I didn't get an email about this issue. Happy to help you look into it. I thought working directory was covered, my build script is also in a separate place to the main projects so I need to override the working directory also.

**Comment by jfernandeze** [comment]
I've tried resolve with thus pull request: https://github.com/cake-contrib/Cake.Karma/pull/9

**Comment by RaringCoder** [comment]
@jfernandeze Yeah I noticed. The working directory is simply applied to the Process, but as you've found that unfortunately support the validation checks beforehand.

**Comment by RaringCoder** [comment]
@jfernandeze Hi mate. I've started a conversation in the gitter channel about this: https://gitter.im/cake-contrib/Lobby

Basically, the working directory argument affects only the Process, but files specified in settings are relative to the Cake script. Cake add-ins tend to take `FilePath` and `DirectoryPath` arguments so you can mix them nicely with the Cake file and directory aliases. In your case, you've supplied the `ConfigFile` argument relative to the working directory you then also provide. I am asking in the channel what the standard behaviour is in this instance. When the command is built, all files should be resolved as absolute paths to remove ambiguity, and the working directory applied to the process itself for the inner command (in this case Karma) to make use of.

**Comment by RaringCoder** [comment]
I'm mistaken, the file paths are not absolute (according to my tests). Not sure that is an issue though as it works for me. I run `build.cake` from src\build with the config file in C:\myapp\src\web\karma.config.js and I set the working directory to C:\myapp\src\web, works as expected.

**Comment by jfernandeze** [comment]
@RaringCoder in your case you put the config file with absolute paths, try use relative paths

**Comment by RaringCoder** [comment]
@jfernandeze Yeah I realised that so I changed it to use relative paths. This also works, but the relative path has to be relative to the script, not the intended working directory of the inner command (ie, Karma). The Cake aliases take `FilePath` and `DirectoryPath` types (not just on the karma aliases), that are "relative path aware", so it breaks convention a little to hack around with the working directory.

I can get our Karma tests to run on a relative path with a working directory without any add-in changes.

**Comment by jfernandeze** [comment]
My problem is that my working directory and my karma.config.js are on the different path... 

My solution was to use the System.IO.Path and Environment:

`		var currentDirectory = Environment.CurrentDirectory;

        var settings = new KarmaStartSettings
        {
           ConfigFile = System.IO.Path.Combine(currentDirectory, "src/Webs/WebSpa/ClientApp/test/karma.conf.js"),		                  
		   RunMode = KarmaRunMode.Local,
		   LocalKarmaCli = System.IO.Path.Combine(currentDirectory, "src/Webs/WebSpa/node_modules/karma-cli/bin/karma"),
		   WorkingDirectory = System.IO.Path.Combine(currentDirectory,"src/Webs/WebSpa")
        };`

**Comment by RaringCoder** [comment]
Hi @jfernandeze,

I see. I will try to reproduce a similar structure myself today and see if there is any improvements that can be made to the Karma runner.
